### PR TITLE
Fix token highlighting for UTF-8 and Claude normalization

### PR DESCRIPTION
**Pull request by shiquda**
## Summary

- switch token highlight range reconstruction from string matching to UTF-8 byte boundary tracking
- add Claude normalization offset handling for full-width punctuation such as `（）`
- add regression tests for UTF-8 boundaries and normalization offset mapping

## Repro

Before this change, token highlighting could break on multibyte characters.
A simple repro was:

```txt
A ❌ B
```

With token highlighting enabled on GPT or Claude, the highlight ranges around `❌` could become misaligned.

## Testing

- `npm test`
- manual verification in VS Code with GPT and Claude highlighting

### CODE REVIEW

File: src/extension.js [hunk 1]
```diff
@@ -123,6 +120,133 @@ function sanitizeProviderSetting(value) {
...
+
+    const segments = [];
+    let index = 0;
+    while (index < text.length) {
+        const codePoint = text.codePointAt(index);
+        const segment = String.fromCodePoint(codePoint);
+        segments.push({ segment, index });
+        index += segment.length;
+    }
+    return segments;
```
Comment (BedirT) [inline src/extension.js]: This fallback walks code points, not graphemes; if it ever took this branch, combining chars and ZWJ emoji would get split across segments, and the check on line 222 would then reject the exact inputs your tests cover (`e\u0301`, `ﬃ`, etc.).

I dug around a bit, and saw that it can't actually trigger with VS Code 1.78+. always has `Intl.Segmenter`, but I'd rather drop it than leave a silently-wrong fallback lying around. Can we throw if Segmenter isn't available instead?

File: src/extension.js [hunk 2]
```diff
@@ -312,54 +436,98 @@ function activate(context) {
...
+        if (byteCursor !== sourceBytes.length) {
+            hasMismatch = true;
         }
 
         editor.setDecorations(tokenDecorations.even, evenRanges);
         editor.setDecorations(tokenDecorations.odd, oddRanges);
+
+        if (hasMismatch) {
+            console.warn('[gpt-token-counter-live] Partial token highlight render due to unsupported UTF-8 token boundaries.');
+        }
```
Comment (BedirT) [inline src/extension.js:559]: Quick flag: this is a behavior change I don't think we want. We have any mismatch in the loop (decode failure, indexOf miss) cleared all highlights and bailed. All-or-nothing; which i think makes sense in an "error" situation.

Now when `hasMismatch` flips mid-loop we `break`, but then still call `setDecorations` with whatever prefix of ranges got built. The `console.warn` is invisible to end users so they'd just see half their doc highlighted and the other half not, with no way to tell what happened.

Can we move the `setDecorations` calls inside an `if (!hasMismatch)` guard, and call `clearTokenHighlights(editor)` in the else branch? Keeps the new byte-level matching but restores predictable all-or-nothing behavior. The warning can stay for debugging.

File: src/extension.js [hunk 3]
```diff
@@ -123,6 +120,133 @@ function sanitizeProviderSetting(value) {
...
+            forward[end] = originalEnd;
+        }
+
+        normalizedOffset = end;
+    }
+
+    const rebuiltNormalizedText = normalizedChunks.join('');
+    if (rebuiltNormalizedText !== normalizedText) {
+        return null;
+    }
```
Comment (BedirT) [inline src/extension.js:218]: One more little thing; can you add a one-line comment explaining what this check is guarding against? NFKC can merge across grapheme boundaries (your `\uFFB5\uFFCC` test is the example), which would make per-grapheme NFKC not reassemble into whole-text NFKC. Not obvious from the code.

**Review by BedirT — COMMENTED** [review]
Thanks for contributing and tackling this! Traced both bugs through the new code and the bug and the solution looks right.

I added two things inline before I merge; one's a code concern, one's a question about intent. Once we tackle those ill pull this in!

**Comment by shiquda** [comment]
Thanks, updated. I removed the non-grapheme fallback and now throw when `Intl.Segmenter` is unavailable, restored all-or-nothing highlight rendering on mismatches, and added a comment explaining the cross-grapheme NFKC reassembly check.
And also added a regression test for the Segmenter requirement and reran `npm test`.
